File: IT/Oggetti/Esercizi/Java/Esercizio3.java
/* file Esercizio3.java */

class Spaceman {
	@Override
	public String toString() {
		return "Sono un astronauta!";
	}
}

public class Esercizio3 {
	public static void main(String [] args) {
		Spaceman neil = new Spaceman();

		// cosa stampa?
		System.out.println(neil);
	}
}

File: IT/1/Oggetti/Esercizi/C++/esercizio3.cpp
/* file esercizio3.cpp */

#include <iostream>

using std::cout;
using std::endl;
using std::ostream;

class Spaceman {
	friend ostream& operator<<(ostream&, const Spaceman&);
};

ostream& operator<<(ostream &os, const Spaceman& spaceman) {
	os << "Sono un astronauta!";
	return os;
}

int main()
{
	Spaceman neil;

	// cosa stampa?
	cout << neil << endl;
}

